src/lib/exports.ts

```diff
@@ -403,3 +426,350 @@ export const exportNoteAsPdfFile = async (
     })
   }
 }
+
+export function getExportAsMenuItems(
+  folderName: string,
+  folderPathname: string,
+  recursive: boolean,
+  startExport: (
+    folderName: string,
+    folderPathname: string,
+    exportType: string,
+    recursive: boolean
+  ) => void
+): MenuItemConstructorOptions[] {
+  return [
+    {
+      type: 'normal',
+      label: 'Export as Markdown (.md)',
+      click: () => startExport(folderName, folderPathname, 'md', recursive),
+    },
+    {
+      type: 'normal',
+      label: 'Export as HTML (.html)',
+      click: () => startExport(folderName, folderPathname, 'html', recursive),
+    },
+    {
+      type: 'normal',
+      label: 'Export as PDF (.pdf)',
+      click: () => startExport(folderName, folderPathname, 'pdf', recursive),
+    },
+  ]
+}
+
+function getPercentage(
+  value: number,
+  maxValue: number,
+  exportProgressMaxValue = 100
+) {
+  if (maxValue === 0) {
+    return 0
+  }
+  return Math.floor((value / maxValue) * exportProgressMaxValue)
+}
+
+async function writeNoteData(
```

Comment:
I don't see any benefits to have this method. It is just writeFile

Reply:
Refactored!